src/Microsoft.ML.TensorFlow/TensorFlow/TensorflowUtils.cs

```diff
@@ -115,8 +115,12 @@ public static IEnumerable<(string, string, ColumnType, string[])> GetModelNodes(
                     Contracts.Assert(metadataType.IsKnownSizeVector && metadataType.ItemType.IsText);
                     schema.GetMetadata(TensorFlowUtils.InputOps, i, ref inputOps);
                 }
-                yield return (name, opType.ToString(), type,
-                    Utils.Size(inputOps.Values) > 0 ? inputOps.Values.Select(input => input.ToString()).ToArray() : new string[0]);
+                var inputOpsValues = inputOps.GetValues();
+                string[] inputOpsResult = new string[inputOpsValues.Length];
+                for (int j = 0; j < inputOpsValues.Length; j++)
+                    inputOpsResult[j] = inputOpsValues[j].ToString();
+
+                yield return (name, opType.ToString(), type, inputOpsResult);
```

Comment:
This code appears to have a bug in it which has been carried over, that is, it appears to assume that the output vector was dense.

Reply:
It appears that the place that populates this metadata always uses a dense vector. See [https://github.com/dotnet/machinelearning/blob/8ca1c9386587aa4ef77c56f605f8244debefce93/src/Microsoft.ML.TensorFlow/TensorFlow/TensorflowUtils.cs#L55](https://github.com/dotnet/machinelearning/blob/8ca1c9386587aa4ef77c56f605f8244debefce93/src/Microsoft.ML.TensorFlow/TensorFlow/TensorflowUtils.cs#L55).

I can add an assert here, if you'd like.  If you want more of a "fix", I can log a bug.  Up to you.

---
In reply to: [232436665](https://github.com/dotnet/machinelearning/pull/1580#discussion_r232436665) [](ancestors = 232436665)

Reply:
How about you just call `DenseValues()` in line 118 and fix the bug?..

---
In reply to: [232822331](https://github.com/dotnet/machinelearning/pull/1580#discussion_r232822331) [](ancestors = 232822331,232436665)